tests/sentry/tasks/test_post_process.py

```diff
@@ -202,6 +203,139 @@ def test_derive_supported_languages(self, mock_derive_code_mappings):
 
             assert mock_derive_code_mappings.delay.call_count == 1
 
+    @patch("sentry.tasks.derive_code_mappings.derive_code_mappings")
+    def test_only_maps_a_given_project_once_per_hour(self, mock_derive_code_mappings):
+        dogs_project = self.create_project()
+        maisey_event = self._create_event(
+            {
+                "fingerprint": ["themaiseymasieydog"],
+            },
+            dogs_project.id,
+        )
+        charlie_event = self._create_event(
+            {
+                "fingerprint": ["charliebear"],
+            },
+            dogs_project.id,
+        )
+        cory_event = self._create_event(
+            {
+                "fingerprint": ["thenudge"],
+            },
+            dogs_project.id,
+        )
+        bodhi_event = self._create_event(
+            {
+                "fingerprint": ["theescapeartist"],
+            },
+            dogs_project.id,
+        )
+
+        self._call_post_process_group(maisey_event)
+        assert mock_derive_code_mappings.delay.call_count == 1
+
+        # second event from project should bail (no increase in call count)
+        self._call_post_process_group(charlie_event)
+        assert mock_derive_code_mappings.delay.call_count == 1
+
+        # advance the clock 59 minutes, and it should still bail
+        with patch("time.time", return_value=time.time() + 60 * 59):
+            self._call_post_process_group(cory_event)
+            assert mock_derive_code_mappings.delay.call_count == 1
+
+        # now advance the clock 61 minutes, and this time it should go through
+        with patch("time.time", return_value=time.time() + 60 * 61):
```

Comment:
TIL

Reply:
Yeah, simpler than I thought it would be. Turns out the cache does expiry just by storing a future timestamp and comparing it to the current time the next time someone asks for the value at that key (at which point it deletes the pair if it's expired).